profiles/schema.py

```diff
@@ -1018,8 +1018,9 @@ def _handle_primary_email(profile, primary_email_input):
 
         profile.emails.exclude(pk=email.pk).filter(primary=True).update(primary=False)
 
-        if not email.primary:
+        if (not email.primary) or (email.verified is not verified):
```

Comment:
I would write this without the parentheses as that's the way it's usually done in python.

```python
if not email.primary or email.verified is not verified:
```

Reply:
Changed.